moxnotify: init at 0.1.0

i-am-logger opened the pull request:
## Description of changes

Adds moxnotify, a feature-rich hardware-accelerated keyboard driven Wayland notification daemon.

### Demo

https://github.com/user-attachments/assets/1996d46c-85cc-4d47-bcf1-5088a58d2192

Features:
- VI bindings for keyboard navigation
- Per-notification configuration
- FreeDesktop Desktop Notifications Specification compliance
- Hardware acceleration via Vulkan
- Sound support via PipeWire
- Notification history
- Lua-based configuration

The package builds both the daemon (`moxnotify`) and control utility (`moxctl`) from the Rust workspace.

## Things done

- [x] Built on platform(s)
  - [x] x86_64-linux
- [x] For non-Linux: Is sandbox = true set in nix.conf? (See Nix manual)
- [x] Tested, as applicable:
  - [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have been tested with the latest nixpkgs-review version
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside nixos/tests)
- [x] Tested compilation of all packages that depend on this change using `nixpkgs-review`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] 24.05 Release Notes (or backporting 23.11 Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (New packages) Added a release notes entry with name, version, and short description
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

## Additional context

This notification daemon is designed for Wayland compositors and provides modern features like hardware acceleration and comprehensive keyboard control. It's a good alternative to mako or dunst for users who want more advanced notification management.

Based on PR #448384 (maintainer update realsnick -> logger).

i-am-logger committed:
moxnotify: init at 0.1.0

Feature-rich hardware-accelerated keyboard driven Wayland notification daemon.

Supports VI bindings, per-notification config, FreeDesktop spec compliance,
hardware acceleration, sound support, and history.

The package builds both daemon and ctl binaries from the workspace,
installs them as moxnotify and moxctl respectively, and includes
the D-Bus service file for proper integration.

Tested with sandboxing enabled on x86_64-linux.

Sigmanificient commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 448405`
Commit: [`df92bf56578a8364ac7793b4a33f6b93baee4db6`](https://github.com/NixOS/nixpkgs/commit/df92bf56578a8364ac7793b4a33f6b93baee4db6) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/df92bf56578a8364ac7793b4a33f6b93baee4db6..pull/448405/head))
Merge: [`82906b1067f019850327134b4a3c10faf2301374`](https://github.com/NixOS/nixpkgs/commit/82906b1067f019850327134b4a3c10faf2301374)

Logs: https://github.com/Sigmanificient/nixpkgs-review-gha/actions/runs/18245446003


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>moxnotify</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>moxnotify</li>
  </ul>
</details>

i-am-logger commented:
@teto i believe i fixed it.

CHN-beta commented on pkgs/by-name/mo/moxnotify/package.nix:
```diff
@@ -0,0 +1,104 @@
+  ];
+
+  # Set LIBCLANG_PATH for bindgen
+  LIBCLANG_PATH = "${libclang.lib}/lib";
```
Maybe better to use `env.LIBCLANG_PATH` ?

i-am-logger replied:
done

i-am-logger commented:
@CHN-beta Thank you for the review feedback! ✅ 

I've updated the package to use `env.LIBCLANG_PATH` instead of the direct assignment, which is indeed the more idiomatic way to set environment variables in nixpkgs.

Changed:
```nix
LIBCLANG_PATH = "${libclang.lib}/lib";
```
To:
```nix
env.LIBCLANG_PATH = "${libclang.lib}/lib";
```

The fix has been committed and should address your concern about proper environment variable handling.

teto commented:
worked for me.
